tests/unit/test_github_session.py

```diff
@@ -156,6 +160,32 @@ def test_token_auth_does_not_use_falsey_values(self):
             s.token_auth(token)
             assert 'Authorization' not in s.headers
 
+    def test_token_auth_with_netrc_works(self):
```

Comment:
I'm wondering why you're setting up `self.tmpdir` for the whole class when we only need it for this test.

Here's the documentation https://docs.pytest.org/en/latest/tmpdir.html

You don't need to import pytest, you can do

```
def test_token_auth_with_netrc_works(self, tmpdir):
    """Verify the behaviour of the library when a token is provided and netrc is present."""
```

Reply:
For some reason that didn't work in the integration tests and I just copied across what did when moving it to a unit test, but seems to work fine for the unit tests.